Src/Newtonsoft.Json/Converters/BinaryConverter.cs

```diff
@@ -152,7 +152,7 @@ private byte[] ReadByteArray(JsonReader reader)
         {
             List<byte> byteList = new List<byte>();
 
-            while (reader.Read())
+            while (reader.ReadAndSkipComments())
```

Comment:
I'd prefer to keep Read() into JsonToken.Comment test in switch statements the way they are. It's slightly faster